### DISCUSSION TITLE
UI 컴포넌트(디자인 시스템)컨벤션 건의

### DISCUSSION THREAD

**Discussion by godhyeongman** [eRentronic/eRentronic-Client#51] [아이디어]
## 주제 
UI 컴포넌트(디자인 시스템) 개발 컨벤션 건의

## 문제
UI 컴포넌트를 재각각 구현하면 변경에 대해선 유리하지만 디자인 통일성 및 개발 편의성이 불리한 부분이 있음
또한 Props의 통일성이 낮아지고 이를 보완하려면 문서를 통해서 지키거나 주의해야함

## 아이디어 
* core라는 범용성 이 높고, 변경에 엄격한 컴포넌트를 만들어 해당 폴더에서는 core를 상속 받아 확장하는 식의 UI 컴포넌트 개발
* core 컴포넌트는 변경이 필요 할 경우? 따로 UI를 만들어 다른 core개발하는 것을 권장함(이는 한 core가 그 컴포넌트의 책임을 보여 주는 것이므로 역할이 늘어나는 것은 바람직하지 못하다 봄)
* 안된다면 팀원에게 사전 공지후, 변경을 리뷰받고 합치는 과정 진행

## 기타
* 현재 core의 존재가 애매한(ex. indicator 폴더,Logo 폴더)경우 core를 안만드는 방식으로 개발 하는 중 임 아이디어 필요

**Comment by herrakam** [comment]
어느정도 비슷한 기능을 가진 공용 컴포넌트들을 core라는 원본을 만들고 이걸 확장하는 방식으로 구현을 하는 방향으로 이해했는데 맞을까요?

**Reply by godhyeongman** [reply]
맞습니다. Button 과 LoadingButton을 완벽하게 따로 개발하는건 비용 낭비 인것 같아서요. 대신 core에 대한 설계 비용이 지불되는것 같아요.

**Reply by herrakam** [reply]
좋은 방법인 것 같습니다! 단지 core를 의무적으로 구현해야 하는 경우에는 문제가 발생할 것 같습니다. 말씀해주셨던 것처럼 core를 만들기 애매한 경우가 문제가 될 거 같은데, core를 만들 각이 보이는 공용 컴포넌트들만 만들어야 할 것 같습니다.

**Reply by godhyeongman** [reply]
'core를 만들 각'에 대한 명확한 기준이 있으면 좋겠는데 정하기 어렵네요. 아이디어 생기시면 discussion에 작성 부탁드리겠습니당 😃
